src/lib/lib3index/cholesky.cc

```diff
@@ -162,11 +163,22 @@ void CholeskyMatrix::compute_row(int row, double* target)
     ::memcpy(static_cast<void*>(target),static_cast<void*>(A_->pointer()[row]),N() * sizeof(double));
 }
 
-CholeskyERI::CholeskyERI(boost::shared_ptr<TwoBodyAOInt> integral, double schwarz,
-    double delta, unsigned long int memory) :
-    integral_(integral), schwarz_(schwarz), Cholesky(delta, memory)
+CholeskyERI::CholeskyERI(boost::shared_ptr<BasisSet> basis, double schwarz,
+    double delta, unsigned long int memory, int num_threads) :
+    basisset_(basis), schwarz_(schwarz), Cholesky(delta, memory)
 {
-    basisset_ = integral_->basis();
+    boost::shared_ptr<IntegralFactory> intfact(new IntegralFactory(basisset_,basisset_,basisset_,basisset_));
```

Comment:
Now if the `basisset_` is not explicitly built as cartesian I (non-deterministically, thread or not threaded) get a segfault that originates from `lib/libmints/twobody.cc:633`. Poking around shows no difference between parallelization here and in other parts of the code. Anyone more familiar with the integral code have ideas? @andysim @jturney